frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -345,40 +345,54 @@ type ConfidenceMetric = {
 };
 
 interface ConfidenceMetricsSectionProps {
-  artifact: Artifact;
+  artifacts: Artifact[];
 }
 
-function ConfidenceMetricsSection({ artifact }: ConfidenceMetricsSectionProps) {
-  const customProperties = artifact.getCustomPropertiesMap();
-  const name = customProperties.get('display_name')?.getStringValue();
+export function ConfidenceMetricsSection({ artifacts }: ConfidenceMetricsSectionProps) {
+  const customPropertiesList = artifacts.map(artifact => artifact.getCustomPropertiesMap());
+  const names = customPropertiesList.map(customProperties =>
+    customProperties.get('display_name')?.getStringValue(),
+  );
 
-  const confidenceMetrics = customProperties
-    .get('confidenceMetrics')
-    ?.getStructValue()
-    ?.toJavaScript();
-  if (confidenceMetrics === undefined) {
+  let confidenceMetricsList = customPropertiesList.map(customProperties =>
+    customProperties
+      .get('confidenceMetrics')
+      ?.getStructValue()
+      ?.toJavaScript(),
+  );
+
+  confidenceMetricsList = confidenceMetricsList.filter(confidenceMetrics => confidenceMetrics);
+  if (confidenceMetricsList.length === 0) {
     return null;
   }
 
-  const { error } = validateConfidenceMetrics((confidenceMetrics as any).list);
+  const rocCurveConfigs: ROCCurveConfig[] = [];
+  for (let i = 0; i < confidenceMetricsList.length; i++) {
+    const confidenceMetrics = confidenceMetricsList[i] as any;
+    const { error } = validateConfidenceMetrics(confidenceMetrics.list);
 
-  if (error) {
-    const errorMsg = 'Error in ' + name + " artifact's confidenceMetrics data format.";
-    return <Banner message={errorMsg} mode='error' additionalInfo={error} />;
+    // If an error exists with confidence metrics, return the first one with an issue.
+    if (error) {
+      const errorMsg = 'Error in ' + names[i] + " artifact's confidenceMetrics data format.";
```

Comment:
Is it possible that the length of array `names` will be different from `rocCurveConfigs`?

Reply:
No - if an item in `rocCurveConfigs` has an item with an error, we return an error banner. Even if a name cannot be found, the `names` array still fills the spot with `undefined`. Therefore, as long as the ROC curves are displayed (in which case they are used), the arrays have the same length.

Reply:
It is fine with either `for` loop or `map`.

My question is more about: You have a logic to filter out any null object in `confidenceMetrics` : `confidenceMetricsList.filter(confidenceMetrics => confidenceMetrics);` How do we make sure the length of `names` and `confidenceMetricsList` equal to each other? I am afraid there can be mismatch due to the filtering of `confidenceMetricsList`

Reply:
Ah you're right, I absolutely missed that I am doing the filtering after already retrieving the `names` - sorry for the mistake, thank you @zijianjoy!

I've updated the logic at https://github.com/kubeflow/pipelines/pull/8046/commits/aad239873499e04f395f044bba22043325db9d3a so the `names` and `confidenceMetrics` are part of the same filtered list, and thus must be the same length. Thank you for pointing this out!

Reply:
It looks much better now, thank you Chris for the fix!